src/wiringx.c

```diff
@@ -400,90 +519,111 @@ EXPORT int wiringXI2CSetup(const char *path, int devId) {
 	return fd;
 }
 
-EXPORT int wiringXSPIGetFd(int channel) {
-	return spi[channel & 1].fd;
+EXPORT int wiringXSPIGetFd(int handle) {
+	struct spi_t *state = NULL;
+
+	if(handle < 0 || handle >= nrspilist) {
+		wiringXLog(LOG_ERR, "wiringX trying to use an unknown SPI connection handle: %i", handle);
+		return -1;
+	}
+
+	state = &spilist[handle].data;
+	return state->fd;
 }
 
-EXPORT int wiringXSPIDataRW(int channel, unsigned char *data, int len) {
+EXPORT int wiringXSPIDataRW(int handle, unsigned char *data, int len) {
+	struct spi_t *state = NULL;
 	struct spi_ioc_transfer tmp;
+
+	if(handle < 0 || handle >= nrspilist) {
+		wiringXLog(LOG_ERR, "wiringX trying to use an unknown SPI connection handle: %i", handle);
+		return -1;
+	}
+
+	state = &spilist[handle].data;
 	memset(&tmp, 0, sizeof(tmp));
-	channel &= 1;
 
 	tmp.tx_buf = (uintptr_t)data;
 	tmp.rx_buf = (uintptr_t)data;
 	tmp.len = len;
-	tmp.delay_usecs = spi[channel].delay;
-	tmp.speed_hz = spi[channel].speed;
-	tmp.bits_per_word = spi[channel].bits_per_word;
+	tmp.delay_usecs = state->delay;
+	tmp.speed_hz = state->speed;
+	tmp.bits_per_word = state->bits_per_word;
 #ifdef SPI_IOC_WR_MODE32
 	tmp.tx_nbits = 0;
 #endif
 #ifdef SPI_IOC_RD_MODE32
 	tmp.rx_nbits = 0;
 #endif
 
-	if(ioctl(spi[channel].fd, SPI_IOC_MESSAGE(1), &tmp) < 0) {
-		wiringXLog(LOG_ERR, "wiringX is unable to read/write from channel %d (%s)", channel, strerror(errno));
+	if(ioctl(state->fd, SPI_IOC_MESSAGE(1), &tmp) < 0) {
+		wiringXLog(LOG_ERR, "wiringX is unable to read/write from SPI device /dev/spidev%u.%u (%s)", spilist[handle].id.device, spilist[handle].id.channel, strerror(errno));
 		return -1;
 	}
 	return 0;
 }
 
-EXPORT int wiringXSPISetup(int channel, int speed) {
-	const char *device = NULL;
-
-	channel &= 1;
+EXPORT int wiringXSPISetup(uint8_t device, uint8_t channel, int speed) {
+	// maximum length of path to spidev: "/dev/spidev255.255\0"
```

Comment:
This assumption isn't checked for.

Reply:
it is countable. I don't understand how this should be checked. We have uint8 for types, and a C string for the path.

Reply:
Also see its use at
snprintf(spidev, sizeof(spidev), "/dev/spidev%u.%u", device, channel);

the function is checked

Reply:
I could add an actual check to catch errors, but this should **never** happen.
```C
if(snprintf(spidev, sizeof(spidev), "/dev/spidev%u.%u", device, channel) >= sizeof(spidev)) {
	wiringXLog(LOG_ERR, "wiringX detected overflow while generating the path to the SPI device node");
	return -1;
}
```

Reply:
that is enforced by uint8. I guess we could include limits.h, and do a conditional pragma error if it doesn't hold true?

Reply:
Why shouldn't you just add a condition that checks the actual value instead of hoping developers read compilation warnings?

Reply:
I'd feel like a fool writing that line of code. But yeah we can do that.

Reply:
> I'd feel like a fool writing that line of code. But yeah we can do that.

But it prevents me having to answer all kinds of questions because people don't RTFM.